### ISSUE TITLE
`invalid-name` false positive on assignment to class attribute `tearDown`

### ISSUE THREAD

**Issue by oprypin** [pylint-dev/pylint#9765]
### Bug description

```python
"""hi"""
import unittest

class BaseCase(unittest.TestCase):
    """hi"""
    def tearDown(self):
        pass

class SubCase(unittest.TestCase):
    """hi"""
    tearDown = BaseCase.tearDown  # [invalid-name] - false positive!
```

### Configuration

```ini
[BASIC]
# https://github.com/google/styleguide/blob/1ae7350b45445b3a3a289c5fe756ab7c80014fe5/pylintrc#L191
class-attribute-rgx=^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$
```

### Command used

```shell
/tmp/pylint/venv/bin/pylint test.py
```

### Pylint output

```
************* Module test
test.py:11:4: C0103: Class attribute name "tearDown" doesn't conform to '^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$' pattern (invalid-name)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)
```

### Expected behavior

```

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

In the past, pylint had this behavior - it did not complain about `invalid-name` here.

a4aff971aeee98365be8b63e146f27e13e7d5873 is the first bad commit

### Pylint version

```shell
pylint 3.3.0-dev0
astroid 3.2.2
Python 3.11.8 (main, May 26 2024, 00:40:16) [GCC 13.2.0]
```

**Comment by jacobtylerwalls** [comment]
Thanks for the bisect!
